src/report.rs

```diff
@@ -0,0 +1,172 @@
+// SPDX-License-Identifier: Apache-2.0
+
+//! Intel SGX Enclave report structures.
+
+use crate::parameters::{Attributes, Features, MiscSelect, Xfrm};
+
+use core::fmt::{self, Display};
+
+#[derive(Debug, PartialEq)]
+#[non_exhaustive]
+pub enum ReportError {
+    InvalidMiscSelect(u32),
+    InvalidFeatures(u64),
+    InvalidXfrm(u64),
+}
+
+impl Display for ReportError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            ReportError::InvalidMiscSelect(value) => {
+                write!(f, "Invalid misc select: {}", value)
+            }
+            ReportError::InvalidFeatures(value) => {
+                write!(f, "Invalid features: {}", value)
+            }
+            ReportError::InvalidXfrm(value) => {
+                write!(f, "Invalid xfrm: {}", value)
+            }
+        }
+    }
+}
```

Comment:
I'm not sure I love this error structure. It is kind of a grab-bag. What if we just had the error be uninterpreted byte array for the value? This would eliminate the need for an error type altogether.